### PR TITLE
Add custom common metric tag feature

### LINKED ISSUE #1707: Provide more default monitoring indicators
### What would you like to be added?

Provide more default monitoring indicators

### Why is this needed?

Provide more default monitoring indicators to make the indicators more comprehensive.

### PR DESCRIPTION
**Pull request by zwmagic**
**What type of PR is this?**

Feature

**What this PR does / why we need it?**

Add custom common metric tag feature

**Which issue(s) this PR fixes？**

Fixes #1707

**Does this PR introduce a user-facing change?**

Yes. We will update user guide at sermant.io.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java [hunk 1]
```diff
@@ -84,6 +87,26 @@ public static Tags of(Map<String, String> tags) {
...
+        tags.put(SCOPE_KEY, SCOPE_CORE);
+        return this;
+    }
+
+    /**
+     * Adds a plugin scope tag.
+     *
+     * @return Tags Returns the current Tags instance to support method chaining.
+     */
+    public Tags addPluginScope() {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: The specific plugin name should be used.

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/config/MetricConfig.java [hunk 2]
```diff
@@ -42,6 +51,12 @@ public class MetricConfig implements BaseConfig {
     @ConfigFieldKey("maxTimeSeries")
     private Integer maxTimeSeries = MAXIMUM_TIME_SERIES_VALUE;
 
+    /**
+     * The metric common tag keys, multiple keys separated by commas
+     */
+    @ConfigFieldKey("common.tag.keys")
+    private String commonTagKeys = String.join(COMMA, MetricCommonTagEnum.getAllKeys());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/config/MetricConfig.java]: The metric.common.tag.keys configuration item is not used here.

File: sermant-agentcore/sermant-agentcore-config/config/config.properties [hunk 3]
```diff
@@ -122,3 +122,5 @@ service.meta.zone=
 metric.type=prometheus
 # The maximum number of metrics.
 metric.maxTimeSeries=1000
+# The metric common tag keys, multiple keys separated by commas
+metric.common.tag.keys=agent,agent.app.name,agent.service.name,agent.ip
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-config/config/config.properties]: The configuration item can be removed.

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java (hunk 1 above)
Comment (zwmagic) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/config/MetricConfig.java (hunk 2 above)
Comment (zwmagic) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/config/MetricConfig.java]: done

File: sermant-agentcore/sermant-agentcore-config/config/config.properties (hunk 3 above)
Comment (zwmagic) [reply sermant-agentcore/sermant-agentcore-config/config/config.properties]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java [hunk 4]
```diff
@@ -1,17 +1,17 @@
 /*
- * Copyright 2017 VMware, Inc.
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: use the original license

Comment (zwmagic) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java [hunk 5]
```diff
@@ -84,6 +86,27 @@ public static Tags of(Map<String, String> tags) {
...
+     * Adds a plugin scope tag.
+     *
+     * @param pluginName plugin name
+     * @return Tags Returns the current Tags instance to support method chaining.
+     */
+    public Tags addPluginScope(String pluginName) {
+        tags.put(SCOPE_KEY, pluginName);
+        return this;
+    }
+
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: give a default scope value as ‘undefined’

Comment (zwmagic) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java [hunk 6]
```diff
@@ -84,6 +92,26 @@ public static Tags of(Map<String, String> tags) {
...
+    /**
+     * Adds a scope tag.
+     *
+     * @param scopeValue scope value
+     * @return Tags Returns the current Tags instance to support method chaining.
+     */
+    public Tags addScope(String scopeValue) {
+        tags.put(SCOPE, StringUtils.isEmpty(scopeValue) ? "undefined" : scopeValue);
+        return this;
+    }
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/api/Tags.java:113]: i suggest to initialize default scope value in constrcutor or MetricCommonTagEnum

**Review by lilai23 — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**